Refresher/Patching/EbootPatcher.cs

```diff
@@ -267,14 +376,41 @@ public void Patch(string url, bool patchDigest)
 
     private static void PatchFromInfoList(BinaryWriter writer, List<PatchTargetInfo> targets, string url, bool patchDigest)
     {
+        Uri uri = new(url);
+        
         byte[] urlAsBytes = Encoding.UTF8.GetBytes(url);
+        byte[] hostAsBytes = Encoding.UTF8.GetBytes(uri.Host);
+        int port = SwapEndianness(uri.Port);
         foreach (PatchTargetInfo target in targets.Where(x => x.Type == PatchTargetType.Url))
         {
             writer.BaseStream.Position = target.Offset;
             writer.Write(urlAsBytes);
 
             //Terminate the rest of the string
-            for (int i = 0; i < target.Length - url.Length; i++) writer.Write('\0');
+            for (int i = 0; i < target.Length - urlAsBytes.Length; i++) writer.Write('\0');
+        }
+
+        foreach (PatchTargetInfo target in targets.Where(x => x.Type == PatchTargetType.Host))
+        {
+            writer.BaseStream.Position = target.Offset;
+            writer.Write(hostAsBytes);
+            
+            //Terminate the rest of the string
+            for (int i = 0; i < target.Length - hostAsBytes.Length; i++) writer.Write('\0');
+        }
+        
+        foreach (PatchTargetInfo target in targets.Where(x => x.Type == PatchTargetType.DoubleNumeric32BitPort))
+        {
+            writer.BaseStream.Position = target.Offset;
+            
+            Debug.Assert(target.Length == sizeof(int) * 2);
+            
+            //TODO: expose somehow the ability to specify separate ports here, this WILL break at some point, as im not sure what the second port is used for yet
```

Comment:
Thats what i'm assuming too, but we dont have an easy way of exposing separate ports right now. 
Given to the best of my research no patches before us have changed the port the game connects on, (and the original server was `10050`, `10051`), should we just increment the second port by 1, and leave a TODO to expose this properly eventually? The existing servers would work with this method, since they will assume port 10050 and 10051

Reply:
I guess, but now I'm left wondering if it's even worth it to change the port in the first place. Cloudflare won't take TCP connections unless you're under a paid plan so you'd be exposing a server to the internet whether you like it or not.

Reply:
> I guess, but now I'm left wondering if it's even worth it to change the port in the first place. Cloudflare won't take TCP connections unless you're under a paid plan so you'd be exposing a server to the internet whether you like it or not.

Given the first port is HTTP, it makes sense to allow changing *that* one (so you can take advantage of cloudflare for that one), but yeah, maybe we should leave the bombd port as is, until people have a good reason to want to change it

Reply:
Yeah, you can change the HTTP port to get CF proxying traffic *I guess*, but you're still going to hit the problem that you can't proxy raw TCP, so bombd won't work. DNS doesn't let you change the destination IP given a certain port.

Again, you're going to have to run a server exposed to the internet, and since other proxies that handle raw TCP are going to let you do arbitrary ports anyways, so I don't see it being worth the effort to patch the port.

Reply:
> Yeah, you can change the HTTP port to get CF proxying traffic _I guess_, but you're still going to hit the problem that you can't proxy raw TCP, so bombd won't work. DNS doesn't let you change the destination IP given a certain port.
> 
> Again, you're going to have to run a server exposed to the internet, and since other proxies that handle raw TCP are going to let you do arbitrary ports anyways, so I don't see it being worth the effort to patch the port.

One reason to want cloudflare on at least the HTTP side i can see is caching of assets, a well-configured server i *think* should be able to tell cloudflare to cache the levels assets and such, which would reduce the load on the servers

Reply:
nv mi thought about it

Reply:
I'm not arguing against using Cloudflare, I'm arguing about the fact that it's impossible for the reasons I described.

Reply:
> nv mi thought about it

lol

Reply:
> I'm not arguing against using Cloudflare, I'm arguing about the fact that it's impossible for the reasons I described.

while there is the chance someone could proxy it with a paid tier, the chance is low, i'll remove the code for now, but keep this in the back of my head, incase i can come up with a patch that allows us to pick separate hosts for bombd and http, as then the code will become relavent again